app/gist/controller.js

```diff
@@ -69,20 +94,69 @@ export default Ember.Controller.extend({
     });
   },
 
+  realNumColumns: computed('numColumns', function() {
+    return Math.min(this.get('numColumns'), MAX_COLUMNS);
+  }),
+  noColumns: computed.equal('numColumns', 0),
+
   /**
-   * Set the initial file columns
+   * Creates the column objects
    */
-  initializeColumns: observer('model', function() {
-    var files = this.get('model.files');
+  createColumns() {
+    let columns = [];
+    for (let i = 0; i < MAX_COLUMNS; ++i) {
+      let col = (i + 1) + "";
+      columns.pushObject(Column.create({
+        col: col,
+        controller: this
+      }));
+    }
+    this.set('columns', columns);
```

Comment:
`createColumns` should likely just be the default implementation of the columns CP getter